src/sagas/flavor.js

```diff
@@ -38,22 +36,140 @@ function* requestStashWorker() {
       throw new Error('Request failed for an unspecified reason!');
     }
   } catch (error) {
-    // eslint-disable-next-line
-    console.dir(error);
-    yield put(actions.requestStashFailure(error));
+    yield put(actions.requestFailure(error));
+  }
+}
+
+function* addStashWorker({ flavor }) {
+  try {
+    const user = yield call(getCurrentUser);
+
+    const endpoint = {
+      url: `/user/${user.id}/flavor`,
+      method: 'POST'
+    };
+
+    const data = {
+      userId: user.id,
+      flavorId: flavor.id
+    };
+
+    const result = yield call(request.execute, { endpoint, data });
+
+    if (result.success) {
+      yield put(actions.updateStashSuccess());
+      yield call(helper.toast, {
+        title: 'Stash Update',
+        message: `Flavor ID ${flavor.id} successfully added!`
+      });
+    } else if (result.error) {
+      throw result.error;
+    } else {
+      throw new Error(`Failed to add Flavor ID ${flavor.id} to the Stash!`);
+    }
+  } catch (error) {
+    const { message } = error;
+
+    yield put(actions.requestFailure(error));
+    yield call(helper.errorToast, message);
+  }
+}
+
+function* removeStashWorker({ flavor }) {
+  try {
+    const user = yield call(getCurrentUser);
+
+    const endpoint = {
+      url: `/user/${user.id}/flavor/${flavor.id}`,
+      method: 'DELETE'
+    };
+
+    const result = yield call(request.execute, { endpoint });
+
+    if (result.success) {
+      yield put(actions.updateStashSuccess());
```

Comment:
Is there any need for a `removeStashSuccess`?

Reply:
I had one originally, to remove any residual artifacts. Currently that is only the flavor note, if I'm not mistaken, and we should probably preserve those.